apps/web/src/components/activity/ExecutionDetail.tsx

```diff
@@ -0,0 +1,112 @@
+import { Container, Grid, useMantineColorScheme } from '@mantine/core';
+import { ExecutionDetailsStatusEnum, StepTypeEnum } from '@novu/shared';
+import { format, parseISO } from 'date-fns';
+import styled from 'styled-components';
+
+import { ExecutionDetailShowRaw } from './ExecutionDetailShowRaw';
+
+import { Button, colors, Text } from '../../design-system';
+import { CheckCircle, Digest, ErrorIcon, Mail, Timer } from '../../design-system/icons';
+
+const DetailText = styled(Text)<{ theme: string }>`
+  color: ${({ theme }) => (theme.colorScheme === 'dark' ? colors.white : colors.B40)};
+  font-size: 14px;
+  font-weight: 400;
+  line-height: 17px;
+`;
+
+const ExecutionDetailStatusWrapper = styled(Container)`
+  padding: 2px;
+  width: 15px;
+`;
+
+const FormattedTime = styled(Text)`
+  color: colors.B60;
+  font-size: 12px;
+  font-height: 16px;
+  padding: 3px 0;
+`;
+
+const getColorByStatus = (theme, status) => {
+  if (status === ExecutionDetailsStatusEnum.FAILED) {
+    return colors.error;
+  }
+
+  if (status === ExecutionDetailsStatusEnum.SUCCESS) {
+    return colors.success;
+  }
+
+  return theme.colorScheme === 'dark' ? colors.B60 : colors.B40;
+};
+
+const getLogoByStatus = (status) => {
+  if (status === ExecutionDetailsStatusEnum.SUCCESS) {
+    return CheckCircle;
+  }
+
+  if (status === ExecutionDetailsStatusEnum.FAILED) {
+    return ErrorIcon;
+  }
+
+  return Timer;
+};
+
+const getLogoByType = (type) => {
+  if (type == StepTypeEnum.DELAY) {
+    return Timer;
+  }
+
+  if (type == StepTypeEnum.DIGEST) {
+    return Digest;
+  }
+
+  return null;
+};
+
+const getExecutionDetailStatus = (status, type) => {
```

Comment:
maybe it is worth to export some of these to a utils file?

Reply:
Let me try to gather them.